ocaml/xapi/xapi_vm.ml

```diff
@@ -256,11 +256,13 @@ let save_uefi_certificates_to_dir ~__context ~self =
     end;
 
   if not (Array.exists (fun e -> e = "PK.auth") !dir) then
-    let file = Unix.openfile "xapi_uefi_certificates.tar" [Unix.O_RDWR; Unix.O_CREAT] 0o755 in
-    let contents = (Xapi_stdext_base64.Base64.decode (Db.Pool.get_uefi_certificates ~__context ~self)) in
-    let hdr = Tar_unix.Header.make "xapi_uefi_certificates.tar" (Int64.of_int (String.length contents)) in
-    Tar_unix.write_block hdr (fun file -> Unixext.really_write_string file contents) file;
-    Tar_unix.Archive.extract (fun _ -> !Xapi_globs.varstore_dir) file
+    begin
+      let file = Unix.openfile "xapi_uefi_certificates.tar" [Unix.O_RDWR; Unix.O_CREAT] 0o755 in
+      let contents = (Xapi_stdext_base64.Base64.decode (Db.Pool.get_uefi_certificates ~__context ~self)) in
+      let hdr = Tar_unix.Header.make "xapi_uefi_certificates.tar" (Int64.of_int (String.length contents)) in
+      Tar_unix.write_block hdr (fun file -> Unixext.really_write_string file contents) file;
```

Comment:
I think these 2 lines may not be required, otherwise you wrap the tar into another tar, and when you extract it you are still left with a tar?

Reply:
but the contents of the uefi_certificates field is a string, and when I'm extracting from a tar file I need a file descriptor?

Reply:
You can save the field to the .tar file directly (without using the tar module, it already is a tar).
There are some helpers in unixext for saving files